src/shims/fs.rs

```diff
@@ -44,27 +43,57 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         let mut options = OpenOptions::new();
 
-        // The first two bits of the flag correspond to the access mode of the file in linux.
+        let o_rdonly = this.eval_libc_i32("O_RDONLY")?;
+        let o_wronly = this.eval_libc_i32("O_WRONLY")?;
+        let o_rdwr = this.eval_libc_i32("O_RDWR")?;
+        // The first two bits of the flag correspond to the access mode in linux, macOS and
+        // windows. We need to check that in fact the access mode flags for the current platform
+        // only use these two bits, otherwise we are in an unsupported platform and should error.
+        if o_rdonly | o_wronly | o_rdwr & !0b11 == 0 {
```

Comment:
```suggestion
        if (o_rdonly | o_wronly | o_rdwr) & !0b11 == 0 {
```
I never know the precedence of these and I think your variant might actually be wrong, as `&` binds more tightly than `|`.

Reply:
That's a good point, me neither.